lib/vf_settokenvalue.cpp

```diff
@@ -550,14 +551,14 @@ namespace ValueFlow
                         setTokenValue(parent, std::move(result), settings);
                     } else if (Token::Match(parent, "%op%")) {
                         if (Token::Match(parent, "%comp%")) {
-                            if (!result.isFloatValue() && !value1.isIntValue() && !value2.isIntValue())
+                            if (!isFloat && !value1.isIntValue() && !value2.isIntValue())
                                 continue;
                         } else {
                             if (value1.isTokValue() || value2.isTokValue())
                                 break;
                         }
                         bool error = false;
-                        if (result.isFloatValue()) {
+                        if (isFloat) {
```

Comment:
Actually, looking at the test, we just need to run calculate as both float if one operand is float and convert to the result type int or float at the end.

Also rather than use lambdas(like intValue1) to do the conversion we should instead update the getIntValue and getFloatValue methods to do the conversion. This should make the code much simpler.

Reply:
This fixes the regression introduced by my change. Thanks.

I will take a look into the remaining false negatives tomorrow. There is also a few test variations I still like to add.

Reply:
It looks like everything has already been fixed. The existing false negatives do not appear to be related to the commit in question but a much earlier one.  Will clean up the tests tomorrow.